qrand/platforms/qsharp/backend.py

```diff
@@ -20,22 +20,37 @@
 ## See the License for the specific language governing permissions and
 ## limitations under the License.
 
+from typing import Optional
+
 from ..backend import QuantumBackend
 
 
 ###############################################################################
 ## QSHARP BACKEND (DECORATOR)
 ###############################################################################
 class QsharpBackend(QuantumBackend):
-    def __init__(self) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
+    def __init__(
+        self,
+        resource_id: Optional[str] = None,
+        target_id: Optional[str] = None,
+    ) -> None:
+        self.resource_id = resource_id
+        self.target_id = target_id
 
     ############################### PUBLIC API ###############################
     @property
     def max_measurements(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+        return 1048576
 
     @property
     def max_qubits(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+        if self.target_id == "ionq.simulator":
+            return 29
+        elif self.target_id == "ionq.qpu":
+            return 11
+        elif self.target_id == "honeywell.hqs-lt-1.0":
+            return 6
+        elif self.target_id == "honeywell.hqs-lt-s1":
+            return 10
+        else:
+            return 1
```

Comment:
Are you going to include support for the local QDK full state simulator? It doesn't cost anything and I think everything in this list would?

Reply:
The local QDK full state simulator will be used by default.

Reply:
Does the local QDK have a `max_qubits = 1` though?

Reply:
No, it's `30`. I've made the changes, just got delayed in pushing it.